tools/dep_updaters/update-openssl.sh

```diff
@@ -10,24 +10,40 @@ cleanup() {
 }
 
 download() {
+
+  NEW_VERSION=$1
+
   if [ -z "$1" ]; then
-    echo "Error: please provide an OpenSSL version to update to"
-    echo "	e.g. ./$0 download 3.0.7+quic1"
-    exit 1
+    NEW_VERSION="$("$NODE" --input-type=module <<'EOF'
+const res = await fetch('https://api.github.com/repos/quictls/openssl/releases');
+if (!res.ok) throw new Error(`FetchError: ${res.status} ${res.statusText}`, { cause: res });
+const releases = await res.json()
+const { tag_name } = releases.at(0);
```

Comment:
This will skip some releases. For instance, when a security patch appears, usually 2 releases are made. In this code you are just selecting the last one (3.x)

Reply:
This pr allows to execute the script with a specified version if not provided will fetch theatest

Reply:
But, this remote call will be executed anyway right? How would it work for OpenSSL 1.x?

Reply:
It will be skipped if provided a specific version in input, thats what we will do for 1.x